### DISCUSSION TITLE
Conditional hiding of form collections

### DISCUSSION THREAD

**Discussion by nicomarg** [jrief/django-formset#172] [Q&A]
I am using df-show/df-hide on fields/fieldsets to hide some stuff dynamically, and was wondering if there was a way to hide FormCollections in the same way.
More precisely, I'm using a form collection with siblings to handle a One-To-Many relation, but this is only useful when some other condition on the form is met, and I would like to hide the whole form collection otherwise.

**Comment by jrief** [comment]
I believe that such a feature can be implemented. However, with a `FormCollection` with siblings there might be some validation issues. I will check if this is possible.

Btw. what is the ue-case where this is needed?

**Comment by nicomarg** [comment]
My use case is to have a single form to create instances of different Child models of the same model (selected based on a ChoiceField). Thus, the relevant fields change dynamically, and one of the child models has a many-to-one relation, which the others don't have. The rest of the data changing from model to model can be handled by Fields, and thus be hidden dynamically, but not the FormCollection handling the many-to-one relation.
I can work around this limitation by asking the user to first create the object, then redirect them to the edit view on the object they just created where they can edit the many-to-one (because at this point we know which child model is being edited, thus we can have different forms for the different child models)

**Comment by Amos-Cacha** [comment]
i would also love to have a feature like this, I have a use case where i need a standalone formcollection to be able to create a model instance with foreign keys in one form (say for example a BookFormCollection that has a BookInfoForm, an AuthorForm, and a PublisherForm as it's attributes, so there is a page where you can just create books), but then in another form collection, i need a field where you can either show a selectize to select an existing instance of the first collection or create a new one using an embedded subform of the first form collection (like a ProfileFormCollection that has a ProfileInfoForm, a SelectOrCreateForm, a SelectBookFieldset and a BookFormCollection as it's attributes so you can create a profile and either select an existing book to display on your profile or create a new one from within the same profile form), so here the SelectOrCreateForm could be some radio buttons or a check box, the SelectBookFieldset could be a Selectize that has a hide_condition dependent on the value of SelectOrCreateForm, so it would be really useful if the BookFormCollection also had a hide condition feature

**Reply by jrief** [reply]
If you wrap it into a `Fieldset`, would that be a solution?

**Reply by Amos-Cacha** [reply]
oh does that work? i wasn't sure whether Fieldset could accept Forms and FormCollections as attributes, ill try it out thanks!

**Reply by jrief** [reply]
no, actually it doesn't. Sorry.
